generator/sbpg/targets/resources/sbp_messages_mod.rs

```diff
@@ -25,11 +25,13 @@ use self::unknown::Unknown;
 #[cfg(feature = "serialize")]
 use serde::{Serialize, Deserialize};
 use crate::serialize::SbpSerialize;
+use crate::framer::FramerError;
 
 pub trait SBPMessage: SbpSerialize {
     fn get_message_type(&self) -> u16;
     fn get_sender_id(&self) -> Option<u16>;
     fn set_sender_id(&mut self, new_id: u16);
+    fn to_frame(&self) -> std::result::Result<Vec<u8>, FramerError>;
```

Comment:
might be able to make this consume the message as well if the original version of that behavior is preferred

Reply:
I'd be in favor of leaving this as is. According to this page of the rust style guide I think an `into_` function like this would be acceptable, but I think the intention is that `into_` should be relatively lightweight if possible. https://doc.rust-lang.org/1.0.0/style/style/naming/conversions.html